Functions/xASL_im_ResampleLinearFair.m

```diff
@@ -1,502 +1,89 @@
-function [output_res]=xASL_im_ResampleLinearFair(im_input,newsize)
-% xASL_im_ResampleLinearFair
+function imOutput = xASL_im_ResampleLinearFair(imInput, newSize)
+%xASL_im_ResampleLinearFair Downsample or upsample a 1D/2D/3D image
 %
-% FORMAT:       [output_res] = xASL_im_ResampleLinearFair(im_input,newsize)
+% FORMAT:       imOutput = xASL_im_ResampleLinearFair(imInput, newSize)
 %
-% INPUT:        im_input    - Image matrix (REQUIRED, DOUBLE, SINGLE or INT)
-%               newsize     - Size of ouput image (REQUIRED, INTEGER ARRAY)
+% INPUT:        imInput     - Image matrix (REQUIRED, DOUBLE, SINGLE or INT)
+%               newSize     - Size of ouput image (REQUIRED, INTEGER ARRAY)
 %
-% OUTPUT:       output_res  - Resampled image matrix (SINGLE)
+% OUTPUT:       imOutput    - Resampled image matrix (SINGLE)
 %
 % -----------------------------------------------------------------------------------------------------------------------------------------------------
-% DESCRIPTION:  Downsample (or upsample, works similarly) old_res image to
-%               low_res image, trilinear.
-%               We recommend using "xASL_spm_Resample" instead, because of
-%               better performance.
-%
-%               {{NB:}} new_res should fit exactly integer fold in old_res
-%
-%               {{NB:}} all dimensions of new_res should have equal size
+% DESCRIPTION:  Downsample or upsample an image from its old to a new resolution.
 %
 % -----------------------------------------------------------------------------------------------------------------------------------------------------
 % EXAMPLE:      image = xASL_io_Nifti2Im('.\M0.nii.gz');
-%               output = xASL_im_ResampleLinearFair(image,[2,2,2]);
+%               output = xASL_im_ResampleLinearFair(image, [2,2,2]);
 %
 % __________________________________
 % Copyright 2015-2021 ExploreASL
```

Comment:
Can you please test with interpolating several image matrices that you get **exactly identical** output, with upscaling and downscaling? I created this function as interpolation methods as SPM etc did degrade the quality of scans by interpolation artifacts, or changing the actual resolution. This was in the context of doing the motion correction in the higher resolution space, where we don't want interpolation artifacts (so no interpolation! just regridding). So please only change my work here if the shortening really gives **identical** results in all cases.

Reply:
Improved unit test: 912f792

Reply:
Does the updated unit test now resample up and down with the old and new technique, and compare both?
I actually thought that the old/slow technique would be the ideal unit test of the newer function Jan wrote.

Reply:
It's not really common to test one script functionality based on another one... This is getting so theoretical. I don't see any benefits of improving this right now. I just wanted to clean up some code and this whole issue got out of hand and I don't want to spend even more time on this ATM.